src/comm_droplet.F

```diff
@@ -510,67 +496,99 @@ subroutine comm_droplet_value(holes_ind,Depart_ind,Depart,Arrive, &
-                                                ! "pdata" are already contiguous
-                                                ! and no action is needed
-             if ( i .eq. num_holes ) exit       ! now the active dropelts in 
-                                                ! the "pdata" are contiguous
-                                                ! after filling the "holes"
-          end if
-       end do
-    end if
-    if(timestats.ge.1) time_dropC4e=time_dropC4e+mytime()
-
-    ! free up the memory for temporary variables
-
-    if (allocated(droplet_n1))  deallocate(droplet_n1)
-    if (allocated(droplet_n2))  deallocate(droplet_n2)
-    if (allocated(droplet_s1))  deallocate(droplet_s1)
-    if (allocated(droplet_s2))  deallocate(droplet_s2)
-    if (allocated(droplet_w1))  deallocate(droplet_w1)
-    if (allocated(droplet_w2))  deallocate(droplet_w2)
-    if (allocated(droplet_e1))  deallocate(droplet_e1)
-    if (allocated(droplet_e2))  deallocate(droplet_e2)
-    if (allocated(droplet_ne1)) deallocate(droplet_ne1)
-    if (allocated(droplet_ne2)) deallocate(droplet_ne2)
-    if (allocated(droplet_nw1)) deallocate(droplet_nw1)
-    if (allocated(droplet_nw2)) deallocate(droplet_nw2)
-    if (allocated(droplet_se1)) deallocate(droplet_se1)
-    if (allocated(droplet_se2)) deallocate(droplet_se2)
-    if (allocated(droplet_sw1)) deallocate(droplet_sw1)
-    if (allocated(droplet_sw2)) deallocate(droplet_sw2)
-
-    if(timestats.ge.1) time_dropC4f=time_dropC4f+mytime()
+     if(timestats.ge.1) time_dropC4e=time_dropC4e+mytime()
+
+     ! free up the memory for temporary variables
+     !$acc end data
+
+     if (allocated(droplet_n1))  deallocate(droplet_n1)
+     if (allocated(droplet_n2))  deallocate(droplet_n2)
+     if (allocated(droplet_s1))  deallocate(droplet_s1)
+     if (allocated(droplet_s2))  deallocate(droplet_s2)
+     if (allocated(droplet_w1))  deallocate(droplet_w1)
+     if (allocated(droplet_w2))  deallocate(droplet_w2)
+     if (allocated(droplet_e1))  deallocate(droplet_e1)
+     if (allocated(droplet_e2))  deallocate(droplet_e2)
+     if (allocated(droplet_ne1)) deallocate(droplet_ne1)
+     if (allocated(droplet_ne2)) deallocate(droplet_ne2)
+     if (allocated(droplet_nw1)) deallocate(droplet_nw1)
+     if (allocated(droplet_nw2)) deallocate(droplet_nw2)
+     if (allocated(droplet_se1)) deallocate(droplet_se1)
+     if (allocated(droplet_se2)) deallocate(droplet_se2)
+     if (allocated(droplet_sw1)) deallocate(droplet_sw1)
+     if (allocated(droplet_sw2)) deallocate(droplet_sw2)
+
+     if(timestats.ge.1) time_dropC4f=time_dropC4f+mytime()
 
   end subroutine comm_droplet_value
 
+!ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+!ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+!ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+
+      subroutine makeContiguous(Depart,Arrive,num_fallout,holes_ind,backfill_ind,pdata,pdata_locind)
```

Comment:
Add some comments about the purpose of this subroutine, as well as if it is GPU-enabled.

Reply:
Does this help?

Reply:
Much better! Thanks for the addition.